scripts/coursera_quiz_conversion.R

```diff
@@ -0,0 +1,79 @@
+# Need here and R.utils
+# For testing use tx_i  <- readLines(here("quizzes", "quiz_ch1.md"))
+
+# This is functionalized but running like script for now till we decide where to put this:
+# run by  pasting `Rscript scripts/coursera_quiz_conversion.R` in the terminal
+
+library(here)
+library(R.utils)
+
+
+coursera_quizzes <- function(path= here("quizzes"), # we might want to update to manuscript
+         verbose = TRUE) {
+  leanpub_quizzes = list.files(
+    pattern = (".md"),
+    ignore.case = TRUE,
+    path = path,
+    full.names = FALSE)
+  if (length(leanpub_quizzes) < 1) {
+    warning("You need quiz files in your specified path dir")
+  }
+  print(leanpub_quizzes)
+  
+  for(i in leanpub_quizzes){
+    tx_i  <- readLines(paste0(path,"/", i))
+   # hashtag <-tx_i[grepl(pattern = "^\\#", tx_i)] # maybe we will need to remove these?
+    ### Remove attempts line
+    tx_i <-tx_i[-grep(pattern = "attempts:", tx_i)]
```

Comment:
I was working on [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)ing up your code a bit and making the logic a bit easier to read using tidyverse, but I don't want to accidentally mess it up either if its for sure working. 

This is what I wrote so far just if you want to see some alternatives. Again some things are just personal preferences. Deleting and adding lines is where things get dicey and not super easy with tidyverse so those lines I don't really have alternatives for. 
```
    a_quiz <- leanpub_quizzes
    
    quiz_content  <- readLines(file.path(path, a_quiz)) 
    
    quiz_df <- data.frame(original_quiz = quiz_content) %>% 
      # Get rid of lines with this
      dplyr::filter(!grepl("Choose the best answer|attempts:", original_quiz)) %>%
      # Replace instances
      dplyr::mutate(updated_quiz = stringr::str_replace(original_quiz, 
                                                      "choose-answers:", 
                                                      "- typeName: multipleChoice"),
                    updated_quiz = gsub("^\\?", "  prompt:", updated_quiz), 
                    updated_quiz = dplyr::case_when(
                      
                      # Where there's a prompt: put a :
                      grepl("prompt:", updated_quiz) ~ gsub(":", "?", updated_quiz), 
                      
                      # Where there's a prompt? put prompt:
                      grepl("prompt?", updated_quiz) ~ gsub("prompt\\?", "prompt:", updated_quiz),
                      
                      # Where it starts with a number, put indent
                      grepl("^[1-9].", updated_quiz) ~ paste0("    ", updated_quiz), 
                      
                      TRUE ~ updated_quiz
                    ), 
                    # Updated how correct answers are shown
                    updated_quiz = gsub("^C\\)", "    - answer:", updated_quiz)) 
```

Reply:
I am going to work on some of these a bit and I will test the code.